### PR TITLE
Adq Patch Inputs step 1

### PR DESCRIPTION
**Pull request by hugo-antoine-rtei**
A first pull request for creating the inputs:
- When creating a new area, the adequacy_patch.ini file will be created containing False (PR#1)
- Add the true/false option in the Optimization Parameters (to use or not the adq patch) (PR#2)

### CODE REVIEW

File: src/libs/antares/study/area/list.cpp [hunk 1]
```diff
@@ -323,6 +328,15 @@ bool saveAreaOptimisationIniFile(const Area& area, const Clob& buffer)
     return ini.save(buffer);
 }
 
+bool saveAreaAdequacyPatchIniFile(const Area& area, const Clob& buffer)
+{
+    IniFile ini;
+    IniFile::Section* section = ini.addSection("adequacy-patch");
+    bool bUseAdequacyPatch = false; //adq: this should be later replaced with area.bUseAdequacyPatch
+    section->add("use-adequacy-patch", static_cast<bool>(bUseAdequacyPatch));
```
Comment (flomnes) [inline src/libs/antares/study/area/list.cpp]: No need to cast, `bUseAdequacyPatch` is a bool.

File: src/libs/antares/study/cleaner/cleaner-v20.cpp [hunk 2]
```diff
@@ -195,6 +195,8 @@ static void PreflightVersion20_area(PathList& e, PathList& p, const Area* area,
     e.add(buffer);
     buffer.clear() << "input/areas/" << id << "/optimization.ini";
     e.add(buffer);
+    buffer.clear() << "input/areas/" << id << "/adequacy_patch.ini";
```
Comment (flomnes) [inline src/libs/antares/study/cleaner/cleaner-v20.cpp:198]: Well done ! This tool us easily forgotten. Adding this here should prevent the study cleaner from removing these newly introduced files.

File: src/libs/antares/study/parameters.cpp [hunk 3]
```diff
@@ -548,6 +549,8 @@ static bool SGDIntLoadFamily_Optimization(Parameters& d,
         return value.to<bool>(d.include.reserve.primary);
     if (key == "include-exportmps")
         return value.to<bool>(d.include.exportMPS);
+    if (key == "include-adequacypatch")
```
Comment (flomnes) [inline src/libs/antares/study/parameters.cpp:553]: Check indentation, use the `clang-format` tool

**Comment by flomnes** [comment]
I'm not sure we need both a global and a local (one for each area) flag. Maybe local flags are enough, the global flag can be deduced from the local flags. The adq patch should be enabled for the simulation if it is set to `true` for any area.

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parameters.cpp (hunk 3 above)
Comment (hugo-antoine-rtei) [reply src/libs/antares/study/parameters.cpp:553]: Should be correct now

File: src/libs/antares/study/area/list.cpp (hunk 1 above)
Comment (hugo-antoine-rtei) [reply src/libs/antares/study/area/list.cpp]: corrected

**Comment by hugo-antoine-rtei** [comment]
I check with Jean-Marc for your comment:
> I'm not sure we need both a global and a local (one for each area) flag. Maybe local flags are enough, the global flag can be deduced from the local flags. The adq patch should be enabled for the simulation if it is set to `true` for any area.
He finds it better to have a global parameter, so as not to have to modify each area when he wants to run the adq patch.

**Pull request closed without merging**